src/native/libs/System.Native/pal_interfaceaddresses.c

```diff
@@ -289,7 +353,16 @@ int32_t SystemNative_GetNetworkInterfaces(int32_t * interfaceCount, NetworkInter
     // To save allocation need for separate free() we will allocate one memory chunk
     // where we first write out NetworkInterfaceInfo entries immediately followed by
     // IpAddressInfo list.
-    void * memoryBlock = calloc((size_t)count, sizeof(NetworkInterfaceInfo));
+#ifdef TARGET_ANDROID
+    // Since Android API 30, getifaddrs returns only AP_INET and AF_INET6 addresses and
+    // we do not get any AP_PACKET addresses and so count == ip4count + ip6count.
+    // We need to make sure that the memoryBlock is large enough to hold all interfaces
+    // and all addresses without any overlap between interfaceList and addressList.
+    int entriesCount = count + ip4count + ip6count;
```

Comment:
I don't fully understand what this means here, if `count == ip4count + ip6count` like the comment says then we're doing `count + count`?

Reply:
Basically yes, the number of different interfaces will be at the most `count` and so the memory block should contain space for `count` times `NetworkInterfaceInfo` and `ip4count + ip6count` for `IpAddressInfo`.

I'll try to revisit the comment to make it clearer.

Reply:
@akoeplinger do you think the comment is clearer now?